eureka/S5_lightcurve_fitting/models.py

```diff
@@ -354,25 +354,67 @@ def eval(self, **kwargs):
         # Generate with batman
         bm_params = batman.TransitParams()
 
+        #check for shared fit
+        # if 'share' in kwargs:
+        #     nchan=kwargs['nchan']
+        #     paramlist=[]
+        #     for param in self.parameters.dict.keys():
+        #         #if 'shared' in self.parameters.dict[param]:
+        #             #for c in np.arange(nchan)-1:
+        #             #    title=param+'_'+str(c+1)
+        #             #    self.parameters.__setattr__(title,self.parameters.dict[param])
+        #             #self.parameters.dict[param]
+        #         if 'free' in self.parameters.dict[param]:
+        #             for c in np.arange(nchan)-1:
+        #                 title=param+'_'+str(c+1)
+        #                 self.parameters.__setattr__(title,self.parameters.dict[param])
+
+                    
+
         # Set all parameters
-        for arg, val in self.parameters.dict.items():
-            setattr(bm_params, arg, val[0])
-        #for p in self.parameters.list:
-        #    setattr(bm_params, p[0], p[1])
+        if 'share' in kwargs:
```

Comment:
I think this will need to change to `if ('share' in kwargs) and (share==True):` once `lc.share` is always passed as an argument.

Reply:
I changed it so that in the model "init" routine it will automatically set share=False if you don't specify share=True, so now it just needs to be if share==True here.
The end result of the changes is that a model initialized with share=True will use shared params, while a model initialized with either share=False or share not specified should result in a fit without shared params.